src/App.js

```diff
@@ -1,7 +1,87 @@
+import React, { useReducer, useEffect } from "react";
 import "./App.css";
+import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
+import Header from "./Component/Header";
+import Home from "./Component/Home";
+import Cart from "./Component/Cart";
+import ProductDetails from "./Component/ProductDetails";
+import PlaceOrder from "./Component/PlaceOrder";
+import SignIn from "./Component/SignIn";
+import Register from "./Component/Register";
+import { auth } from "./firebase";
+
+export const CartContext = React.createContext();
+
+const initialState = {
+  product: [],
+  search: "",
+  user: "",
+};
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "addToCart":
+      return {
+        ...state,
+        product: [...state.product, action.value.productDetails],
+      };
+    case "removeFromCart":
+      let newCart = [...state.product];
+      const index = state.product.findIndex((item) => (item.id = action.id));
+      if (index >= 0) {
+        newCart.splice(index, 1);
+      }
```

Comment:
Actually, I have tried that first but the problem was if  I add the same item more than one time in the cart, and then when trying to remove them, instead of removing only that item it was removing all the items with the same id...
// code 
 return {
        ...state,
        product: state.product.filter((ele) => ele.id !== action.id),
      };

Reply:
actually, if one item is added more than once, then we should locate that in the state and increment qty against that id. Then for every inc/dec operation we should adjust qty.